examples/glium_gl_area/main.rs

```diff
@@ -0,0 +1,280 @@
+use std::{cell::RefCell, ptr, rc::Rc};
+
+use glium::{
+    implement_vertex, index::PrimitiveType, program, uniform, Frame, IndexBuffer, Surface,
+    VertexBuffer,
+};
+use gtk::{gdk, glib::WeakRef, prelude::*};
+
+struct GLAreaBackend {
+    // GLArea will transitively hold a strong reference to this struct, so to prevent a reference
+    // cycle, this struct must hold a weak reference to the GLArea.
+    area: WeakRef<gtk::GLArea>,
+}
+
+unsafe impl glium::backend::Backend for GLAreaBackend {
+    fn swap_buffers(&self) -> Result<(), glium::SwapBuffersError> {
+        // GLArea handles this by itself.
```

Comment:
Does this maybe require signalling to the `GLArea` that its dirty and needs to redraw?

Reply:
The intention is that you only do this within the `render` signal. I'm not sure what happens if you do it outside.

Reply:
Makes sense but maybe add a comment about that then :)